File: r/grains/grains.R
# Name: grains.R
# Purpose: exercism, R track, grains
#     The function square takes an integer 1-64, for squares on the chess board
#       and returns the power 2^(n-1).
#     The function total tells how many grains it takes to fill the board.
# Written by Z Knight, 2019.10.14
#
square <- function(n) {
  if (n < 1 || n > 64) {
    stop("Input outside of chessboard range 1-64. Exiting.")
  } else if (n == 1) {
    return(1)
  } else {
    return(2*square(n-1))
  }
}

total <- function() {
  square(64)*2-1
}

File: python/grains/grains.py
#! /usr/bin/env python
"""
    Name:
        grains.py
    Purpose:
        exercism, python track, grains: 
            Calculate the number of grains of wheat on a chessboard given that the number on each square doubles.
    Written by:
        Z Knight, 2021.11.06
"""

def square(number):
    if number < 1 or number > 64:
        print("error")
        raise ValueError("square must be between 1 and 64")
        return 0
    if number == 1:
        return 1
    return 2*(square(number-1))


def total():
    return square(64)*2-1
